--- include/tests_kernel.old	2014-09-23 12:11:02.000000000 -0700
+++ include/tests_kernel	2014-09-23 16:27:30.000000000 -0700
@@ -489,12 +489,14 @@
                 if [ "${FIND}" = "" ]; then
                      logtext "Result: No kernels (/boot/vmlinuz*) found on disk"
                      ReportException "${TEST_NO}:2" "Can not find any vmlinuz files in /boot, which is unexpected"
-                  else
+                else
                      # Replace dashes to allow numeric sort
                      MYKERNEL=`uname -r | sed 's/\.[a-z].*.//g' | sed 's/-[a-z].*.//g' | sed 's/-/./g'`
                      logtext "Result: using ${MYKERNEL} as my kernel version (stripped)"
                      # Display kernels, extract version numbers and sort them numeric per column (up to 6 numbers)
-                     KERNELS=`ls /boot/vmlinuz* | sed 's/vmlinuz-//' | sed 's/\.[a-z].*.//g' | sed 's/-[a-z].*.//g' | sed 's./boot/..' | sed 's/-/./g' | sort -n -k1,1 -k2,2 -k3,3 -k4,4 -k5,5 -k6,6 -t \.`
+                     KERNELS=`ls /boot/{vmlinuz,kernel}* | sed 's/\(vmlinuz\|kernel[a-zA-Z0-9-]*\)-//' | sed 's/\.[a-z].*.//g' | sed 's/-[a-z].*.//g' | sed 's./boot/..' | sed 's/-/./g' | sort -n -k1,1 -k2,2 -k3,3 -k4,4 -k5,5 -k6,6 -t \.`
+		     # Testing only.
+		     #echo "${KERNELS}" > /root/Lynis/kernels.txt
                      FOUND_KERNEL=0
                      for I in ${KERNELS}; do
                          if [ ${FOUND_KERNEL} -eq 1 ]; then
